src/main/java/org/jabref/logic/xmp/XmpUtilWriter.java

```diff
@@ -159,6 +171,26 @@ private static void writeDublinCore(PDDocument document,
         catalog.setMetadata(metadataStream);
     }
 
+    public static String generateXmpString(List<BibEntry> entries) {
+        XMPMetadata meta = XMPMetadata.createXMPMetadata();
+        for (BibEntry entry : entries) {
+            DublinCoreSchema dcSchema = meta.createAndAddDublinCoreSchema();
+            XmpUtilWriter.writeToDCSchema(dcSchema, entry, null, Globals.prefs.getXMPPreferences());
+        }
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        XmpSerializer serializer = new XmpSerializer();
+        try {
+            serializer.serialize(meta, os, true);
+            return os.toString(StandardCharsets.UTF_8.name());
+        } catch (TransformerException e) {
+            LOGGER.warn("Tranformation into xmp not possible: " + e.getMessage());
```

Comment:
Why not logging the exception itself, too?

Reply:
Do you mean the stack trace of the exception?
Is there a smart solution in JabRef to print the stack trace with the logger (maybe an utility method)?

Otherwise, I would use something like this:
Arrays.asList(exception.getStackTrace())
                .stream()
                .map(Objects::toString)
                .collect(Collectors.joining("\n"))

Reply:
The logger methods have  a second parameter which accepts a throwable /exception.
The stack trace will then automatically be printed (handled by the logger itself)

Reply:
The code usually looks something like `LOGGER.warn("Transformation in xmp not possible", e)`.